auth_core/pkce.py

```diff
@@ -0,0 +1,50 @@
+import base64
+import hashlib
+import httpx
+import logging
+import secrets
+
+from urllib.parse import urljoin
+
+from .token_data import TokenData
```

Comment:
```suggestion
from . import token_data
```

for easier test mocks in general.